src/plugins/ietf/ietf_interface.cpp

```diff
@@ -0,0 +1,530 @@
+    char *if_name = nullptr;
+    sr_change_iter_t *iter = nullptr;
+    sr_change_oper_t op = SR_OP_CREATED;
+    sr_val_t *old_val = nullptr;
+    sr_val_t *new_val = nullptr;
+    sr_xpath_ctx_t xpath_ctx = { 0, };
+    int rc = SR_ERR_OK, op_rc = SR_ERR_OK;
+    std::shared_ptr<interface> intf;
+
+    SRP_LOG_INF("In %s", __FUNCTION__);
+
+    /* no-op for apply, we only care about SR_EV_{ENABLED,VERIFY,ABORT} */
+    if (SR_EV_APPLY == event)
+        return SR_ERR_OK;
+
+    SRP_LOG_DBG("'%s' modified, event=%d", xpath, event);
+
+    /* get changes iterator */
+    rc = sr_get_changes_iter(session, xpath, &iter);
+    if (SR_ERR_OK != rc) {
+        sr_free_change_iter(iter);
+        SRP_LOG_ERR("Unable to retrieve change iterator: %s", sr_strerror(rc));
+        return rc;
+    }
+
+    foreach_change (session, iter, op, old_val, new_val) {
+
+        SRP_LOG_DBG("A change detected in '%s', op=%d",
+                    new_val ? new_val->xpath : old_val->xpath, op);
+        if_name = sr_xpath_key_value(new_val ? new_val->xpath : old_val->xpath,
+                                     "interface", "name", &xpath_ctx);
+
+        intf = interface::find(if_name);
+        if (nullptr == intf) {
+            SRP_LOG_ERR_MSG("Interface does not exist");
+            return SR_ERR_INVAL_ARG;
+        }
+
+        switch (op) {
+            case SR_OP_CREATED:
+            case SR_OP_MODIFIED:
+                if (new_val->data.bool_val)
+                    intf->set(interface::admin_state_t::UP);
+                else
+                    intf->set(interface::admin_state_t::DOWN);
+                break;
+            case SR_OP_DELETED:
+                intf->set(interface::admin_state_t::DOWN);
+                break;
+            default:
+                break;
+        }
+        sr_xpath_recover(&xpath_ctx);
+
+        sr_free_val(old_val);
+        sr_free_val(new_val);
+
+        /* Commit the changes to VOM DB and VPP */
+        if ( OM::write("enable/"+intf->key(), *intf) != rc_t::OK ) {
+            SRP_LOG_ERR_MSG("Fail writing changes to VPP");
+            return SR_ERR_OPERATION_FAILED;
+        }
+
+    }
+    sr_free_change_iter(iter);
+
+    return SR_ERR_OK;
+}
+
+static int
+ipv46_config_add_remove(const std::string &if_name,
+                        const std::string &addr, uint8_t prefix,
+                        bool add)
+{
+    std::shared_ptr<l3_binding> l3;
+    std::shared_ptr<interface> intf;
+    rc_t rc = rc_t::OK;
+
+    intf = interface::find(if_name);
+    if (nullptr == intf) {
+        SRP_LOG_ERR_MSG("Interface does not exist");
+        return SR_ERR_INVAL_ARG;
+    }
+
+    try {
+
+        VOM::route::prefix_t pfx(addr, prefix);
+        l3 = std::make_shared<l3_binding>(*intf, pfx);
+
+        #define KEY(l3) "l3_" + l3->itf().name() + "_" + l3->prefix().to_string()
+        if (add) {
+            /* Commit the changes to VOM DB and VPP */
+            if ( OM::write(KEY(l3), *l3) != rc_t::OK ) {
```

Comment:
Add comments saying:
Work for modification too because OM::write check existing instances